lib/block_view_stake.go

```diff
@@ -0,0 +1,2600 @@
+	if err != nil {
+		return errors.Wrapf(err, "LockedStakeEntry.Decode: Problem reading LockedAtEpochNumber: ")
+	}
+
+	// ExtraData
+	lockedStakeEntry.ExtraData, err = DecodeExtraData(rr)
+	if err != nil {
+		return errors.Wrapf(err, "LockedStakeEntry.Decode: Problem reading ExtraData: ")
+	}
+
+	return err
+}
+
+func (lockedStakeEntry *LockedStakeEntry) GetVersionByte(blockHeight uint64) byte {
+	return 0
+}
+
+func (lockedStakeEntry *LockedStakeEntry) GetEncoderType() EncoderType {
+	return EncoderTypeLockedStakeEntry
+}
+
+//
+// TYPES: StakeMetadata
+//
+
+type StakeMetadata struct {
+	ValidatorPublicKey *PublicKey
+	StakeAmountNanos   *uint256.Int
+}
+
+func (txnData *StakeMetadata) GetTxnType() TxnType {
+	return TxnTypeStake
+}
+
+func (txnData *StakeMetadata) ToBytes(preSignature bool) ([]byte, error) {
+	var data []byte
+	data = append(data, EncodeByteArray(txnData.ValidatorPublicKey.ToBytes())...)
+	data = append(data, EncodeUint256(txnData.StakeAmountNanos)...)
+	return data, nil
+}
+
+func (txnData *StakeMetadata) FromBytes(data []byte) error {
+	rr := bytes.NewReader(data)
+
+	// ValidatorPublicKey
+	validatorPublicKeyBytes, err := DecodeByteArray(rr)
+	if err != nil {
+		return errors.Wrapf(err, "StakeMetadata.FromBytes: Problem reading ValidatorPublicKey: ")
+	}
+	txnData.ValidatorPublicKey = NewPublicKey(validatorPublicKeyBytes)
+
+	// StakeAmountNanos
+	txnData.StakeAmountNanos, err = DecodeUint256(rr)
+	if err != nil {
+		return errors.Wrapf(err, "StakeMetadata.FromBytes: Problem reading StakeAmountNanos: ")
+	}
+
+	return nil
+}
+
+func (txnData *StakeMetadata) New() DeSoTxnMetadata {
+	return &StakeMetadata{}
+}
+
+//
+// TYPES: UnstakeMetadata
+//
+
+type UnstakeMetadata struct {
+	ValidatorPublicKey *PublicKey
+	UnstakeAmountNanos *uint256.Int
+}
+
+func (txnData *UnstakeMetadata) GetTxnType() TxnType {
+	return TxnTypeUnstake
+}
+
+func (txnData *UnstakeMetadata) ToBytes(preSignature bool) ([]byte, error) {
+	var data []byte
+	data = append(data, EncodeByteArray(txnData.ValidatorPublicKey.ToBytes())...)
+	data = append(data, EncodeUint256(txnData.UnstakeAmountNanos)...)
+	return data, nil
+}
+
+func (txnData *UnstakeMetadata) FromBytes(data []byte) error {
+	rr := bytes.NewReader(data)
+
+	// ValidatorPublicKey
+	validatorPublicKeyBytes, err := DecodeByteArray(rr)
+	if err != nil {
+		return errors.Wrapf(err, "UnstakeMetadata.FromBytes: Problem reading ValidatorPublicKey: ")
+	}
+	txnData.ValidatorPublicKey = NewPublicKey(validatorPublicKeyBytes)
+
+	// UnstakeAmountNanos
+	txnData.UnstakeAmountNanos, err = DecodeUint256(rr)
+	if err != nil {
+		return errors.Wrapf(err, "UnstakeMetadata.FromBytes: Problem reading UnstakeAmountNanos: ")
+	}
+
+	return nil
+}
+
+func (txnData *UnstakeMetadata) New() DeSoTxnMetadata {
+	return &UnstakeMetadata{}
+}
+
+//
+// TYPES: UnlockStakeMetadata
+//
+
+type UnlockStakeMetadata struct {
+	ValidatorPublicKey *PublicKey
+	StartEpochNumber   uint64
+	EndEpochNumber     uint64
```

Comment:
See my comment in the _connect. I think we can simplify and just have this function unlock ALL unlockable stake when you call it. Hard to imagine someone wanting anything other than this.